src/hooks/index.js

```diff
@@ -116,35 +145,49 @@ exports.hasSchoolPermission = (inputPermission) => async (context) => {
 
 	const { schoolId: school } = user;
 
-	const results = await Promise.allSettled(user.roles.map(async (role) => {
-		const { permissions = {} } = school;
-		// If there are no special school permissions, continue with normal permission check
-		if (!permissions[role.name]
-				|| !Object.prototype.hasOwnProperty.call(permissions[role.name], inputPermission)) {
-			return hasPermission(inputPermission)(context);
-		}
-		// Otherwise check for user's special school permission
-		if (permissions[role.name][inputPermission]) {
-			return context;
-		}
-		throw new Forbidden(`You don't have one of the permissions: ${inputPermission}.`);
-	}));
+	const results = await Promise.allSettled(
+		user.roles.map(async (role) => {
+			const { permissions = {} } = school;
+			// If there are no special school permissions, continue with normal permission check
+			if (
+				!permissions[role.name] ||
```

Comment:
Codacy found an issue: ['||' should be placed at the beginning of the line.](https://app.codacy.com/gh/hpi-schul-cloud/schulcloud-server/pullRequest?prid=6045787)